compiler/rustc_typeck/src/check/expr.rs

```diff
@@ -805,7 +829,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         self.check_lhs_assignable(lhs, "E0070", span);
 
         let lhs_ty = self.check_expr_with_needs(&lhs, Needs::MutPlace);
-        let rhs_ty = self.check_expr_coercable_to_type(&rhs, lhs_ty, Some(lhs));
+        let rhs_ty =
+            self.check_expr_coercable_to_type(&rhs, lhs_ty, Some(lhs), TypeAscriptionCtxt::Normal);
```

Comment:
this is kind of a contradiction -- check that it's coercable to a type, but not a coercion site? but I think this parameter should be removed anyway.